extensions/parquet/table/src/main/java/io/deephaven/parquet/table/transfer/ObjectArrayAndVectorTransfer.java

```diff
@@ -10,21 +10,38 @@
  import org.jetbrains.annotations.NotNull;
 
  import java.util.Arrays;
+ import java.util.function.Supplier;
 
 /**
  * Used as a base class of arrays/vectors of transfer objects for types like strings or big integers that need
  * specialized encoding.
+ * @param <T> The type of the data in the column, could be an array/vector
+ * @param <V> The type of the values in the array/vector
  */
-abstract class ObjectArrayAndVectorTransfer<T> extends ArrayAndVectorTransfer<T, Binary[], Binary[]> {
-    private int bufferSize;
+abstract class ObjectArrayAndVectorTransfer<T, V> extends ArrayAndVectorTransfer<T, Binary[], Binary[]> {
+    /**
+     * Number of values added to the buffer
+     */
     private int bufferedDataCount;
+    /**
+     * Total number of bytes buffered
+     */
     private int numBytesBuffered;
 
+    /**
+     * Used as a temporary buffer for storing encoded data for a single row before it is copied to the main buffer.
+     * Allocated lazily because of the high cost of construction of Binary objects.
```

Comment:
It's not a `Binary`. It's an array of references to `Binary`. Not sure why this one buffer needs to be allocated lazily.